tritium-tracing/src/main/java/com/palantir/tritium/tracing/RemotingCompatibleTracingInvocationEventHandler.java

```diff
@@ -38,6 +42,44 @@
 
     private static final Logger logger = LoggerFactory.getLogger(RemotingCompatibleTracingInvocationEventHandler.class);
 
+    private static final AtomicBoolean shouldLogFallbackError = new AtomicBoolean();
+
+    private static final Supplier<Boolean> requiresFallback = Suppliers.memoize(() -> {
+        try {
+            return isUsingMultipleTracers(loadRemoting3TracersClass());
+        } catch (ReflectiveOperationException e) {
+            // expected case when remoting3 is not on classpath
+            return false;
+        }
+    });
+
+    private static final Supplier<Tracer> tracerFactory = Suppliers.memoize(() -> {
+        // Ugly reflection based check to determine what implementation of tracing to use to avoid duplicate
+        // traces as remoting3's tracing classes delegate functionality to tracing-java in remoting 3.43.0+
+        // and remoting3.tracing.Tracers.wrap now returns a "com.palantir.tracing.Tracers" implementation.
+        //
+        // a) remoting3 prior to 3.43.0+ -> use reflection based remoting3 tracer
+        // b) remoting3 3.43.0+ -> use tracing-java
+        // c) no remoting3 -> use tracing-java
```

Comment:
At what point do you think we can remove remove the remoting3 implementation?

Reply:
I'd love to remove it, but there are some services with remoting3 (and even remoting2) on classpaths that need to be upgraded first.

Reply:
@felixdesouza atlasdb has fully moved off remoting2/3 clients right? All the other usages internally seemed to be just ObjectMappers.

Reply:
So we still reference remoting2 `RemoteException` and a few other error decoder related classes for our "custom" feign proxy stuff. We are in the experimental phases of running pure conjure-java-runtime for our clients, and when we get full confidence, we can remove all of that and we'll have no references to it. This is likely on the order of a couple of weeks. @jeremyk-91 @jkozlowski for SA.

Reply:
I think it's only the `remoting3 prior to 3.43.0` case that we're worried about here, right? Are we aware of any services that satisfy:
1. Use remoting3 prior to 3.43.0
2. Take Tritium upgrades
3. Use tracing

I'm pretty confident that the issue which lead to this PR was in an environment that cannot take advantage of tracing, regardless of the implementation.